giskard/core/suite.py

```diff
@@ -123,6 +155,68 @@ def _repr_html_(self):
         widget = TestSuiteResultWidget(self)
         return widget.render_html()
 
+    def upload(self, client: GiskardClient, label: Optional[str] = None):
+        if self.suite.id is None:
+            raise ValueError("Cannot save suite execution result of unsaved suite")
+
+        client.save_test_suite_execution_result(self.suite.id, self.suite.project_key, self._to_dto(label))
+
+    def _to_dto(self, label: Optional[str]) -> SaveSuiteExecutionDTO:
+        return SaveSuiteExecutionDTO(
+            label=label or self.execution_date.isoformat().replace("T", " ").split(".")[0],
+            suiteId=self.suite.id,
+            inputs=TestInputDTO.from_inputs_dict(self.inputs),
+            result=TestSuiteExecutionResult.PASSED if self.passed else TestSuiteExecutionResult.FAILED,
+            message="",
+            results=[self._test_result_to_dto(result) for result in self.results],
+            executionDate=self.execution_date.isoformat(),
+            completionDate=self.completion_date.isoformat(),
+        )
+
+    @staticmethod
+    def _test_result_to_dto(result: SuiteResult):
+        datasets = {dataset.id: dataset for dataset in result.params.values() if isinstance(dataset, Dataset)}
+
+        return SaveSuiteTestExecutionDTO(
```

Comment:
How do we have so many field 😱

Reply:
I can work work on this on this PR it's relatively safe to clean this up